packages/client/src/workflow-client.ts

```diff
@@ -451,6 +457,33 @@ export interface IntoHistoriesOptions {
   bufferLimit?: number;
 }
 
+/**
+ * Defines how to start a workflow when using {@link WorkflowClient.startUpdateWithStart} and
+ * {@link WorkflowClient.executeUpdateWithStart}. `workflowIdConflictPolicy` is required in the options.
+ */
+export class WithStartWorkflowOperation<T extends Workflow> {
+  private _workflowHandle: Promise<WorkflowHandle<T>>;
+  private _resolveWorkflowHandle!: (handle: WorkflowHandle<T>) => void;
+  public _executed = false;
+
```

Comment:
Don't prefix private propsA few things here:

1. Let's avoid this "underscore prefix" convention if we can. In TS, `_xxx` usually means an unused variable/parameter (e.g. to silence linters). Using that as a visibility marker is unusual. And yes, I know this file contains many such occurences, but that's something we should depart from, not replicate.

2. You can expose the handle as `public readonly workflowHandle`, so you'll no longer need a method below.

3. Use symbols to define protected methods/fields; for example:

```
const resolveWorkflowHandle: unique symbol = Symbol();
const operationExecuted: unique symbol = Symbol();

export class WithStartWorkflowOperation<T extends Workflow> {
  public readonly workflowHandle: Promise<WorkflowHandle<T>>;
  public [operationExecuted]: boolean = false;

  constructor(
    public workflowTypeOrFunc: string | T,
    public options: WorkflowStartOptions<T> & { workflowIdConflictPolicy: WorkflowIdConflictPolicy }
  ) {
    this.workflowHandle = new Promise<WorkflowHandle<T>>((resolve) => {
      this[resolveWorkflowHandle] = resolve;
    });
  }
}
```

Reply:
Also, we should probably `reject` the `WorkflowHandle` promise if the Workflow failed to start.

Reply:
Thanks, great suggestions here.

> You can expose the handle as `public readonly workflowHandle`, so you'll no longer need a method below.

We'll still want a callable won't we? I.e. this would look odd:


```typescript
const wfHandle = await startOp.workflowHandle
```

Reply:
Resolved as per group decision to use a promise-returning `.workflowhandle()`, reworked class